lms/djangoapps/instructor/static/instructor/ProblemBrowser/components/Main/Main.test.jsx

```diff
@@ -1,96 +1,99 @@
                     fetchCourseBlocks={fetchCourseBlocksMock}
                     problemResponsesEndpoint={problemResponsesEndpoint}
-                    onSelectBlock={jest.fn()}
-                    selectedBlock="some-selected-block"
+                    onSelectBlock={onSelectBlockMock}
+                    selectedBlock={props.selectedBlock}
                     taskStatusEndpoint={taskStatusEndpoint}
+                    reportDownloadEndpoint={reportDownloadEndpoint}
+                    ShowBtnUi="false"
+                    {...props}
                 />
             </Provider>,
-        );
-        // eslint-disable-next-line prefer-destructuring
-        const instance = component.root.children[0].instance;
-        instance.handleToggleDropdown();
-        expect(fetchCourseBlocksMock.mock.calls.length).toBe(1);
+        )
+    );
+
+    describe('Initial rendering', () => {
+        test('should match snapshot with basic parameters', () => {
+            const { container } = renderMainComponent();
+            expect(container).toMatchSnapshot();
+        });
+
+        test('should match snapshot with selected block', () => {
+            const { container } = renderMainComponent({ selectedBlock: 'some-selected-block' });
+            expect(container).toMatchSnapshot();
+        });
     });
 
-    test('display dropdown on toggling dropdown', () => {
-        const component = shallow(
-            <Main
-                courseId={courseId}
-                createProblemResponsesReportTask={jest.fn()}
-                excludeBlockTypes={excludedBlockTypes}
-                fetchCourseBlocks={jest.fn()}
-                problemResponsesEndpoint={problemResponsesEndpoint}
-                onSelectBlock={jest.fn()}
-                selectedBlock="some-selected-block"
-                taskStatusEndpoint={taskStatusEndpoint}
-            />,
-        );
-        expect(component.find(BlockBrowserContainer).length).toBeFalsy();
-        component.find(Button).find({ label: 'Select a section or problem' }).simulate('click');
-        expect(component.find(BlockBrowserContainer).length).toBeTruthy();
+    describe('Dropdown interactions', () => {
+        test('should fetch course blocks when dropdown is toggled', async () => {
+            renderMainComponent();
+
+            await userEvent.click(screen.getByText('Select a section or problem'));
+
+            await waitFor(() => {
+                expect(fetchCourseBlocksMock).toHaveBeenCalledTimes(1);
+                expect(fetchCourseBlocksMock).toHaveBeenCalledWith(courseId, excludedBlockTypes);
+            });
+        });
+
+        test('should display dropdown when toggled', async () => {
+            renderMainComponent();
+            expect(screen.queryByTestId('mocked-block-browser-container')).toBeNull();
+            await act(async () => {
+                await userEvent.click(screen.getByText('Select a section or problem'));
+            });
```

Comment:
Does this need to be wrapped in `await act`?

What happens when this is just
```suggestion
            await userEvent.click(screen.getByText('Select a section or problem'));
```

Reply:
Yes it could be, I have tried it and it does not give errors. Also remove act on import